### DISCUSSION TITLE
JSX within Markdown links

### DISCUSSION THREAD

**Discussion by dimaip** [mdx-js/mdx#1609] [Q&A] [answered]
This currently doesn't work in V2. Any plan to support things like this?

```
[click here]({props.href})
```

We're using MDX as a templating engine for emails, and it would be much easier for editors to author links using markdown, rather than having to type:

```
<a href={props.href}>click here</a>
```

**Comment by ChristianMurphy** [comment, marked as answer]
Similar discussion in XDM, which is merging back with MDX https://github.com/wooorm/xdm/issues/88

**Reply by dimaip** [reply]
Honestly I'd love curly braces to be interpolated in as many places as possible. If I needed a literal value somewhere I could always escape it.
Making non tech-savvy people type the full <a> tag format is not cool and kinda inconsistent.

**Reply by wooorm** [reply]
mdx is javascript and JSX. With a few shortcuts. If you care about non tech-savvy users, I would highly recommend a normal template language.

**Reply by dimaip** [reply]
Thanks, I know what MDX is :-)
I have been using MDX on a bunch of projects with non-developers being in charge of editing MDX files. It worked amazing! The ability to have Markdown with sprinkles of React components in the right places adds incredible power to the format with which people are well familiar.
E.g. we were using MDX as a format for email templates and it has been a real blessing.
